### PR TITLE
alightTryCatch indent rule option

### PR DESCRIPTION
**Pull request by sbcgua**
to #337 

P.S. I'd apply `createRunner` to all the test in the file, would you mind ?
P.P.S. what is your TS flow ? straight forward approach seems to be very bulky: run `tsc -w` in one console, run `mocha --watch` in another ... and running debugger is pain ... maybe I need to read a couple of articles on how to develop for ts ;)

### CODE REVIEW

**Comment by larshp** [comment]
tooling, yeah, it needs to be looked into, I've not had any luck setting up the watch options properly

File: test/abap/pretty_printer.ts [hunk 1]
```diff
@@ -3,6 +3,17 @@ import {PrettyPrinter} from "../../src/abap/pretty_printer";
 import {MemoryFile} from "../../src/files";
 import {Registry} from "../../src/registry";
 
+const testTitle = (text: string): string => { return text.split("\n")[0]; };
+const createRunner = (test: any, methodName: string): (() => void) => {
```
Comment (larshp) [inline test/abap/pretty_printer.ts]: this is a function?

Comment (larshp) [reply test/abap/pretty_printer.ts]: only used in one place

File: test/abap/pretty_printer.ts [hunk 2]
```diff
@@ -3,6 +3,17 @@ import {PrettyPrinter} from "../../src/abap/pretty_printer";
...
+  return () => {
+    const reg = new Registry().addFile(new MemoryFile("zfoo.prog.abap", test.input)).parse();
+    expect(reg.getABAPFiles().length).to.equal(1);
+    const printer = new PrettyPrinter(reg.getABAPFiles()[0], test.options);
+    const result = (printer as any)[methodName]();
```
Comment (larshp) [inline test/abap/pretty_printer.ts]: this looks fishy, it breaks all assisted refactoring and where-used

File: test/abap/pretty_printer.ts (hunk 1 above)
Comment (sbcgua) [reply test/abap/pretty_printer.ts]: Yes, but the proposal is to apply to all suites in this test - the code is almost same. But have a look below comment - it is all connected.

Comment (sbcgua) [reply test/abap/pretty_printer.ts]: removed

File: test/abap/pretty_printer.ts (hunk 2 above)
Comment (sbcgua) [reply test/abap/pretty_printer.ts]: removed

**Comment by sbcgua** [comment]
started to write a long text why I did this and what was the goal and then just rewritten the test same way as others ... it is just the test after all ;)

**Comment by larshp** [comment]
There are some utility methods/functions somewhere that can replace

```
      const reg = new Registry().addFile(new MemoryFile("zfoo.prog.abap", test.input)).parse();
      expect(reg.getABAPFiles().length).to.equal(1);
      const result = new PrettyPrinter(reg.getABAPFiles()[0]).getExpectedIndentation(); 
```

with just one line of code, but I think lets just keep it out of the scope for this PR, its just nice to have, currently its easy to understand what happens, even though there is too much code

**Pull request merged**